### ISSUE TITLE
Add integration with rust-jemalloc-pprof for easy mem perf

### ISSUE THREAD

**Issue by jiacai2050** [apache/horaedb#1577]
### Describe This Problem

[rust-jemalloc-pprof](https://github.com/polarsignals/rust-jemalloc-pprof) is a cool project to do memory perf.

### Proposal

Add a new HTTP endpoint `/debug/pprof/heap` to serve the profiling.

**Comment by zuston** [comment]
Maybe I can pick up this. I have some experience about tikv jemalloc crate to dump

**Comment by jiacai2050** [comment]
@zuston Thanks, assigned.

**Comment by zuston** [comment]
> @zuston Thanks, assigned.

From this crate doc, I found this way is not better than the vaillina jeprof way. It just generates the pprof format, if you want to dig it with human format, you have to execute in the host machine instead of any remote machine to call api directly. Because it still depends on the execute binary.

**Comment by zuston** [comment]
PTAL @jiacai2050

**Comment by jiacai2050** [comment]
> you have to execute in the host machine instead of any remote machine to call api directly.

What do you mean by call api directly?

With pprof, we can explore the perf data in various format, such as [flamegraph](https://github.com/google/pprof/issues/401#issuecomment-1088744271)
- https://github.com/google/pprof/blob/main/doc/README.md#graph

**Comment by vmandke** [comment]
@jiacai2050 is this still available to pick ?

**Comment by MichaelLeeHZ** [comment]
您好：    邮件已收到！

**Comment by jiacai2050** [comment]
@vmandke Yeah, it's still available, but you should merge the PR to https://github.com/apache/horaedb/tree/analytic-engine branch.

**Comment by jiacai2050** [comment]
Hi, thanks for you interests.

The new API should also contains a `seconds` params, sorry for misleading.

**Comment by BaldDemian** [comment]
I made a PR [here](https://github.com/apache/horaedb/pull/1627), but there is a problem about the usage of the returned `heap.pb.gz` file🤔. You can see https://github.com/apache/horaedb/pull/1627#issuecomment-2880772777 for more description.

**Comment by jiacai2050** [comment]
👍 

I will review it in next couple of days.
